src/client/pmix_client_get.c

```diff
@@ -564,23 +564,24 @@ static void _getnbfn(int fd, short flags, void *cbdata)
      * that was provided at startup */
     if (0 == strncmp(cb->key, "pmix", 4)) {
         /* should be in the internal hash table. */
+        if (PMIX_SUCCESS != (rc = pmix_hash_fetch(&nptr->internal, cb->rank, cb->key, &val))) {
 #if defined(PMIX_ENABLE_DSTORE) && (PMIX_ENABLE_DSTORE == 1)
-        if (PMIX_SUCCESS == (rc = pmix_dstore_fetch(cb->nspace, cb->rank, cb->key, &val))) {
+            if (PMIX_SUCCESS != (rc = pmix_dstore_fetch(cb->nspace, cb->rank, cb->key, &val))) {
+                goto request;
+            }
 #else
-        if (PMIX_SUCCESS == (rc = pmix_hash_fetch(&nptr->internal, cb->rank, cb->key, &val))) {
+            goto request;
 #endif
-            /* found it - we are in an event, so we can
-             * just execute the callback */
-            cb->value_cbfunc(rc, val, cb->cbdata);
-            /* cleanup */
-            if (NULL != val) {
-                PMIX_VALUE_RELEASE(val);
-            }
-            PMIX_RELEASE(cb);
-            return;
         }
-        /* if we don't have it, go request it */
-        goto request;
+        /* found it - we are in an event, so we can
+         * just execute the callback */
+        cb->value_cbfunc(rc, val, cb->cbdata);
+        /* cleanup */
+        if (NULL != val) {
+            PMIX_VALUE_RELEASE(val);
+        }
+        PMIX_RELEASE(cb);
+        return;
     }
 
     /* otherwise, the data must be something they "put" */
```

Comment:
As far as I recall the the key name that caused particular error for OMPI was "pmix.loc". But what if the key with some arbitrary name was submitted through PMIx_Store_local?